### PR TITLE
Breaking down files to MVC structure

### PR DESCRIPTION
**Pull request by Daniel-Shwartzman**
Removed amount column, created mvc structure with basic functions for controller and view, modified disco.py to run the new changes.

### CODE REVIEW

File: disko.py [hunk 1]
```diff
@@ -1,4 +1,14 @@
-from src.disko.image_collector import ImageCollector
+from src.mvc.view import ImageRegistryManager
```
Comment (dpointk) [inline disko.py]: let's leave the structure with src.disko.xx.xxx as it's important for future integration. @Daniel-Shwartzman 
also change the mvc folder to something else.

File: src/mvc/controller.py
Comment (dpointk) [inline src/mvc/controller.py:1]: The file name is controller.py , please change to image_controller.py

File: src/mvc/model.py
Comment (dpointk) [inline src/mvc/model.py:1]: also here , please change to image_model.py

File: src/mvc/view.py
Comment (dpointk) [inline src/mvc/view.py:1]: change filename to image_view.py

File: disko.py [hunk 2]
```diff
@@ -1,4 +1,14 @@
...
-collector = ImageCollector()
-collector.collect_images()
+# main function
+def main():
+    # Database file path
+    db_file = 'image_data.db'
+
+    # Run GUI
+    gui = ImageRegistryManager(db_file)
+    gui.run()
```
Comment (dpointk) [inline disko.py]: There's a different approach we would like to achieve.
The way it is built in your commit now, is that the controlling of what happens is I the view. It should be in the controller.
The controller should gather all the information and send it to the view.
The way you have structured it here is that the view is in charge. We should change it.
The idea is that the view should be replaceable tomorrow. So no logic should happen there. All should happen in the controller.

File: src/disko/image_management/image_view.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        # Update column display based on checkbox selection
+        selected_columns = [column.get() for column in self.checkboxes.values()]
+        for col in self.columns:
+            if col in selected_columns:
+                self.treeview.column(col, display=True)
+            else:
+                self.treeview.column(col, display=False)
+    
+   
+    def run(self):
```
Comment (dpointk) [inline src/disko/image_management/image_view.py]: This function should get all the required data from the controller which it I turn should run it.
The main python script will run a function in the controller which will decide how to output the data.

File: src/disko/image_management/image_view.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+            self.treeview.heading(col, text=col)  # Set column headings
+
+        self.treeview.pack(side='left', fill='both', expand=True)
+
+        # Add scrollbar to the treeview
+        scrollbar = ttk.Scrollbar(frame, orient='vertical', command=self.treeview.yview)
+        scrollbar.pack(side='right', fill='y')
+        self.treeview.configure(yscrollcommand=scrollbar.set)
+
+        cluster_names = self.controller.get_kubernetes_clusters()
```
Comment (dpointk) [inline src/disko/image_management/image_view.py]: This should be given as a parameter from the controller.

**Review by dpointk — CHANGES_REQUESTED** [review]
Please check my comments in the view and main section

**Comment by Daniel-Shwartzman** [comment]
We would like your help, we cannot entirley resolve the import situation, this is the error we get with the current commit below:
Traceback (most recent call last):
  File "/home/daniel/Desktop/disko/disko.py", line 2, in <module>
    from src.disko.image_management.image_controller import ImageController
  File "/home/daniel/Desktop/disko/src/disko/image_management/image_controller.py", line 4, in <module>
    from src.disko.image_management.image_view import ImageRegistryManager
  File "/home/daniel/Desktop/disko/src/disko/image_management/image_view.py", line 4, in <module>
    from src.disko.image_management.image_controller import ImageController
ImportError: cannot import name 'ImageController' from partially initialized module 'src.disko.image_management.image_controller' (most likely due to a circular import) (/home/daniel/Desktop/disko/src/disko/image_management/image_controller.py)

**Comment by dpointk** [comment]
You're not supposed to import the image controller in the view.
The only function of the view is to print the output.
I think the fact that you're using this tk to manage buttons and manage the application is the issue.
You need to figure a way to setup the gui and reset it using the controller

**Comment by dpointk** [comment]
See this answer and comment about PySide

https://stackoverflow.com/questions/38042632/mvc-the-simplest-example

**Comment by Daniel-Shwartzman** [comment]
To fix the import error we removed the cluster selection option, so the scanning happens on the current cluster.
The view does not contain the controller, the controller runs the view.

**Pull request closed without merging**